### PR TITLE
Fix compilation and IDE inspection warnings in dependency-management

### PR DESCRIPTION
**Pull request by ljacomet**
This is quite a large set of changes, a number using auto fix from the IDE, validated by the Java compiler and test execution.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RFM

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultArtifactIdentifier.java [hunk 1]
```diff
@@ -81,23 +84,19 @@ public boolean equals(Object o) {
 
         DefaultArtifactIdentifier that = (DefaultArtifactIdentifier) o;
 
-        if (classifier != null ? !classifier.equals(that.classifier) : that.classifier != null) {
-            return false;
-        }
-        if (extension != null ? !extension.equals(that.extension) : that.extension != null) {
+        if (!Objects.equals(classifier, that.classifier)) {
```
Comment (melix) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultArtifactIdentifier.java:87]: :x: We should not introduce Guava `Objects`. While convenient, it has the drawback of creating arrays for every comparison, which turns out to create a lot of garbage collection.

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Pull request merged**